dspace/modules/api/src/main/java/org/dspace/workflow/WorkflowEmailManager.java

```diff
@@ -113,17 +115,24 @@ private static void notifyOfApproval(Context c, Item i, String emailFilename)
             }else{
                 email.addArgument("");
             }
-            String manuScriptIdentifier = "";
-            DCValue[] manuScripIdentifiers = i.getMetadata(MetadataSchema.DC_SCHEMA, "identifier", "manuscriptNumber", Item.ANY);
-            if(0 < manuScripIdentifiers.length){
-                manuScriptIdentifier = manuScripIdentifiers[0].value;
+            String manuscriptIdentifier = "";
+            DCValue[] manuscriptIdentifiers = i.getMetadata(MetadataSchema.DC_SCHEMA, "identifier", "manuscriptNumber", Item.ANY);
+            if(0 < manuscriptIdentifiers.length){
+                manuscriptIdentifier = manuscriptIdentifiers[0].value;
             }
 
-	    if(manuScriptIdentifier.length() == 0) {
-		manuScriptIdentifier = "none available";
-	    }
+            if(manuscriptIdentifier.length() == 0) {
+                manuscriptIdentifier = "none available";
+            }
+            email.addArgument(manuscriptIdentifier);
 
-            email.addArgument(manuScriptIdentifier);
+            // add a DOI URL as well:
+            String doi_url = "";
+            Matcher doimatcher = Pattern.compile("doi:(.+)").matcher(datapackageDoi);
+            if (doimatcher.find()) {
```

Comment:
should log an error if the DOI is not found, and put replacement text in the email argument